platforms/chibios/drivers/wear_leveling/wear_leveling_legacy_config.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Nick Brassel (@tzarc)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+// Work out the page size to use
+#ifndef WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE
+#    if defined(QMK_MCU_STM32F042)
+#        define WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE 1024
+#    elif defined(QMK_MCU_STM32F070) || defined(QMK_MCU_STM32F072)
+#        define WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE 2048
+#    elif defined(QMK_MCU_STM32F401) || defined(QMK_MCU_STM32F411)
+#        define WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE 16384
+#    endif
+#endif
+
+// Work out how much flash space we have
+#ifndef WEAR_LEVELING_LEGACY_EMULATION_FLASH_SIZE
+#    define WEAR_LEVELING_LEGACY_EMULATION_FLASH_SIZE ((*(uint32_t *)FLASHSIZE_BASE) & 0xFFFFU) // in kB
+#endif
+
+// The base location of program memory
+#ifndef WEAR_LEVELING_LEGACY_EMULATION_FLASH_BASE
+#    define WEAR_LEVELING_LEGACY_EMULATION_FLASH_BASE 0x08000000
+#endif
+
+// The number of pages to use
+#ifndef WEAR_LEVELING_LEGACY_EMULATION_PAGE_COUNT
+#    if defined(QMK_MCU_STM32F042)
+#        define WEAR_LEVELING_LEGACY_EMULATION_PAGE_COUNT 2
+#    elif defined(QMK_MCU_STM32F070) || defined(QMK_MCU_STM32F072)
+#        define WEAR_LEVELING_LEGACY_EMULATION_PAGE_COUNT 1
+#    elif defined(QMK_MCU_STM32F401) || defined(QMK_MCU_STM32F411)
+#        define WEAR_LEVELING_LEGACY_EMULATION_PAGE_COUNT 1
+#    endif
+#endif
+
+// The origin of the emulated eeprom
+#ifndef WEAR_LEVELING_LEGACY_EMULATION_BASE_PAGE_ADDRESS
+#    if defined(QMK_MCU_STM32F042) || defined(QMK_MCU_STM32F070) || defined(QMK_MCU_STM32F072)
+#        define WEAR_LEVELING_LEGACY_EMULATION_BASE_PAGE_ADDRESS ((uintptr_t)(WEAR_LEVELING_LEGACY_EMULATION_FLASH_BASE) + WEAR_LEVELING_LEGACY_EMULATION_FLASH_SIZE * 1024 - (WEAR_LEVELING_LEGACY_EMULATION_PAGE_COUNT * WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE))
+#    elif defined(QMK_MCU_STM32F401) || defined(QMK_MCU_STM32F411)
+#        if defined(BOOTLOADER_STM32)
+#            define WEAR_LEVELING_LEGACY_EMULATION_BASE_PAGE_ADDRESS (WEAR_LEVELING_LEGACY_EMULATION_FLASH_BASE + (1 * (WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE))) // +16k
+#        elif defined(BOOTLOADER_TINYUF2)
+#            define WEAR_LEVELING_LEGACY_EMULATION_BASE_PAGE_ADDRESS (WEAR_LEVELING_LEGACY_EMULATION_FLASH_BASE + (3 * (WEAR_LEVELING_LEGACY_EMULATION_PAGE_SIZE))) // +48k
```

Comment:
Just to be on the same page. TinyUF2 occupies the first 31kb flash, so we could start at a 32kb offset giving us 32kb flash in total to work until QMK starts.

Reply:
I'd specifically placed it at 48kB just in case they end up producing an over-sized tinyuf2 binary. It's able to be overridden if someone _needs_ 32kB of backing store, but I don't imagine that'd be necessary in all cases.
The current `eeprom_stm32` implementation uses only 16kB for backing store, at any rate... we're not constraining things further in this PR.